test/functional/feature_pruning.py

```diff
@@ -494,5 +499,19 @@ def test_scanblocks_pruned(self):
         assert_raises_rpc_error(-1, "Block not available (pruned data)", node.scanblocks,
             "start", [{"desc": f"raw({false_positive_spk.hex()})"}], 0, 0, "basic", {"filter_false_positives": True})
 
+    def test_pruneheight_undo_presence(self):
+        node = self.nodes[2]
+        pruneheight = node.getblockchaininfo()["pruneheight"]
+        fetch_block = node.getblockhash(pruneheight - 1)
+
+        self.connect_nodes(1, 2)
+        peers = node.getpeerinfo()
+        peer_id = peers[0]["id"]
+        node.getblockfrompeer(fetch_block, peer_id)
```

Comment:
nit: `peer_id` makes things less readable imo
```suggestion
        node.getblockfrompeer(fetch_block, peers[0]["id"])
```

Reply:
done